### PR TITLE
utils/cpu definition change

### PR DESCRIPTION
**Pull request by tfauck**
change online_list() to online_cpu_list()
change online_count() to online_cpu_count()
ange get_faily() to get_cpu_arch()

Signed-off-by: Thierry <tfauck@free.fr>

### CODE REVIEW

File: generic/stressapptest.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ def setUp(self):
         self.test_file = self.params.get('tmp_file', default='/tmp/dummy')
         self.duration = self.params.get('duration', default='30')
         self.threads = self.params.get(
-            'threads', default=cpu.online_count())
+            'threads', default=cpu.online_cpus__count())
```
Comment (harish-24) [inline generic/stressapptest.py:30]: `cpu__count` typo

File: cpu/em_cpuidle.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ def test(self):
                 val = process.system_output("cat /sys/devices/system/cpu/"
                                             "cpu%s/cpuidle/state%s/"
                                             "name" % (cpu_num, i)).decode("utf-8")
-                if 'power8' in cpu.get_family():
+                if 'power8' in cpu.get_cpu_arch():
```
Comment (harish-24) [inline cpu/em_cpuidle.py:71]: This change is wrong. cpu arch is `powerpc` whereas family is `power8`.

File: kernel/kernbench.py [hunk 3]
```diff
@@ -122,7 +122,7 @@ def setUp(self):
         self.kernel_version = platform.uname()[2]
         self.iterations = self.params.get('runs', default=1)
         self.threads = self.params.get(
-            'cpus', default=2 * cpu.online_count())
+            'cpus', default=2 * cpu.online_cpus__count())
```
Comment (harish-24) [inline kernel/kernbench.py:125]: here too

File: toolchain/gcc.py [hunk 4]
```diff
@@ -45,6 +45,10 @@ def setUp(self):
             packages.extend(['libmpfr-dev', 'libgmp-dev', 'libmpc-dev',
                              'texinfo', 'zip', 'libc6-dev', 'libelf1',
                              'elfutils', 'gnat', 'autogen'])
+        elif dist.name == 'debian':
```
Comment (harish-24) [inline toolchain/gcc.py:48]: I see only few difference between `Ubuntu` and `debian`. Can you add debian list as base list and extend left overs like `'texinfo', 'gnat'` for Ubuntu under same condition?

File: toolchain/oprofile.py [hunk 5]
```diff
@@ -35,9 +35,11 @@ def setUp(self):
             deps.append("git-core")
         else:
             deps.append("git")
-        if detected_distro.name in ["Ubuntu", 'debian']:
-            deps.append("libxml2-utils")
-            deps.append("tclsh")
+        if detected_distro.name in ["Ubuntu"]:
+            deps.append(["libxml2-utils", "tclsh", "oprofile"])
```
Comment (harish-24) [inline toolchain/oprofile.py:39]: here also you can extend package name for Ubuntu.
```
if detected_distro.name in ["Ubuntu", "debian"]:
    deps.extend(["libxml2-utils", "tclsh"])
    if detected_distro.name == "Ubuntu":
        deps.extend(["oprofile"])
```

And also note than the `append` is wrong here. It has to be `extend` if you are using a list of packages.

File: toolchain/papiTest.py [hunk 6]
```diff
@@ -65,6 +65,7 @@ def test(self):
         errors = 0
         warns = 0
         for line in result.stdout.splitlines():
```
Comment (harish-24) [inline toolchain/papiTest.py:67]: https://github.com/avocado-framework-tests/avocado-misc-tests/pull/1758#discussion_r433057601

File: fuzz/fsfuzzer.py [hunk 7]
```diff
@@ -81,7 +81,7 @@ def setUp(self):
         self._args = self.params.get('fstype', default='')
         self._fsfuzz = os.path.abspath(os.path.join('.', "fsfuzz"))
         fs_sup = process.system_output('%s %s' % (self._fsfuzz, ' --help'))
-        match = re.search(br'%s' % self._args, fs_sup, re.M | re.I)
+        match = re.search(br'%s' % self._args, fs_sup.decode(), re.M | re.I)
```
Comment (harish-24) [inline fuzz/fsfuzzer.py:84]: `b` must be removed here?

Comment (tfauck) [reply fuzz/fsfuzzer.py:84]: No - I have tested all combinations and we need to keep the br'%s' for search to be properly done.

Comment (harish-24) [reply fuzz/fsfuzzer.py:84]: can you paste a sample output with matches the search?

Comment (tfauck) [reply fuzz/fsfuzzer.py:84]: 1) Currently Reproduced traceback from: /usr/local/lib/python3.7/site-packages/avocado_framework-78.0-py3.7.egg/avocado/core/test.py:854
Traceback (most recent call last):
  File "/home/fedora/avocado-fvt-wrapper/tests/avocado-misc-tests/fuzz/fsfuzzer.py", line 84, in setUp match = re.search(br'%s' % self._args, fs_sup, re.M | re.I)
TypeError: %b requires a bytes-like object, or an object that implements __bytes__, not 'str'
ERROR 1-tests/avocado-misc-tests/fuzz/fsfuzzer.py:Fsfuzzer.test -> TestSetupFail: %b requires a bytes-like object, or an object that implements __bytes__, not 'str'

2) If I Add a debug message, and get rid of 'b' 
[stdout] DEBUG: self.args= fs_sup=b'./fsfuzz cramfs|ext2|ext3|ext4|swap|btrfs|iso9660|msdos|squashfs|vfat|xfs'
[stdout] 


Reproduced traceback from: /usr/local/lib/python3.7/site-packages/avocado_framework-78.0-py3.7.egg/avocado/core/test.py:854
Traceback (most recent call last):
  File "/home/fedora/avocado-fvt-wrapper/tests/avocado-misc-tests/fuzz/fsfuzzer.py", line 85, in setUp
    match = re.search(r'%s' % self._args, fs_sup, re.M | re.I)
  File "/usr/lib64/python3.7/re.py", line 183, in search
    return _compile(pattern, flags).search(string)
TypeError: cannot use a string pattern on a bytes-like object

3) If I add decode to fs_sup then It sounds to work so you are right, my appologize for not having it right first time - thanks for the review.
[stdout] DEBUG: self.args= fs_sup=b'./fsfuzz cramfs|ext2|ext3|ext4|swap|btrfs|iso9660|msdos|squashfs|vfat|xfs' <class 'bytes'>
[stdout]

**Pull request merged**